pkg/kubelet/cm/cpumanager/topology/topology.go

```diff
@@ -51,6 +52,15 @@ func (topo *CPUTopology) CPUsPerCore() int {
 	return topo.NumCPUs / topo.NumCores
 }
 
+// CPUsPerUncoreCache returns the average number of logical CPUs are associated with
+// each uncore cache id. Even CPUs share the same llc id may not the same.
+func (topo *CPUTopology) CPUsPerUncoreCache() int {
+	if topo.NumUnCoreCaches == 0 {
+		return 0
+	}
+	return topo.NumCPUs / topo.NumUnCoreCaches
```

Comment:
In some special cases, it may give wrong answer. In VM for example, host machine may reserve cores to for virtualization and these cores are reserved evenly on multiple chips, in these case, kubelet in VM gets all the caches same as the host machine, but the threads are less than host machine, then CPUsPerUncoreCache will be wrong.

Reply:
thanks, good question, the key is not evenly allocate of physical cpu to vcpu. this depend too much on the virtualization implementation of cloud vendors, and in some caces(like yours) the problem occurs.
Although in our cloud the problem is not exists, but, i will take it to consideration and make a change, to make the algorithm more common.